### ISSUE TITLE
hx-swap="scroll:none" not working on a progressively enhanced form

### ISSUE THREAD

**Issue by sjc5** [bigskysoftware/htmx#1947]
Prior discussion here: https://github.com/bigskysoftware/htmx/pull/1459#issuecomment-1791420162

Goal: Prevent scroll-to-top behavior after hx-boosted form submissions.

Code that should achieve this per the linked discussion above:

```
<body hx-swap="innerHTML" hx-boost="true">
  <form method="POST" action="/whatever" hx-swap="scroll:none">
```

Result: Still scrolls to top.

I believe this is a bug in HTMX and will do some digging in / debugging to find out.

Thanks much

**Comment by sjc5** [comment]
Ok, I see what the issue is.

Under the hood, `hx-show` calls `element.scrollIntoView`, and `hx-scroll` calls `element.scrollTop`.

Boosted elements get `show: top` by default. So you don't want to override `hx-scroll`, you want to override `hx-show`.

I think this just needs a documentation PR to (1) explain what `hx-show` and `hx-scroll` actually mean in terms of the real DOM APIs, and (2) explain that if you want to override it for a boosted element, you need to use `hx-show`, not `hx-scroll`.

If you agree happy to pull a PR.

**Comment by xhaggi** [comment]
You are right. I added a note in the `hx-swap` documentation on how to disable the default behavior on a per-element basis in #1459.

**Comment by sjc5** [comment]
Addressed in #1459
